de.metas.business/src/main/java/org/adempiere/pricing/interceptor/M_DiscountSchemaBreak.java

```diff
@@ -0,0 +1,175 @@
+package org.adempiere.pricing.interceptor;
+
+import java.math.BigDecimal;
+import java.util.stream.Stream;
+
+import org.adempiere.ad.modelvalidator.annotations.Interceptor;
+import org.adempiere.ad.modelvalidator.annotations.ModelChange;
+import org.adempiere.bpartner.service.IBPartnerDAO;
+import org.adempiere.exceptions.AdempiereException;
+import org.adempiere.pricing.api.CalculateDiscountRequest;
+import org.adempiere.pricing.api.IEditablePricingContext;
+import org.adempiere.pricing.api.IMDiscountSchemaBL;
+import org.adempiere.pricing.api.IPricingBL;
+import org.adempiere.pricing.limit.PriceLimitRuleContext;
+import org.adempiere.pricing.limit.PriceLimitRuleResult;
+import org.adempiere.util.Services;
+import org.compiere.model.I_M_DiscountSchemaBreak;
+import org.compiere.model.ModelValidator;
+import org.springframework.stereotype.Component;
+
+import de.metas.product.IProductBL;
+import lombok.NonNull;
+
+/*
+ * #%L
+ * de.metas.business
+ * %%
+ * Copyright (C) 2018 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+@Interceptor(I_M_DiscountSchemaBreak.class)
+@Component
+public class M_DiscountSchemaBreak
+{
+	private static final String MSG_UnderLimitPriceWithExplanation = "UnderLimitPriceWithExplanation";
+
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW, ModelValidator.TYPE_BEFORE_CHANGE })
+	public void beforeSave(final I_M_DiscountSchemaBreak schemaBreak)
+	{
+		// If M_Product_ID and M_Product_Category_ID is set, M_Product_ID has priority - teo_sarca [ 2012439 ]
+		if (schemaBreak.getM_Product_ID() > 0 && schemaBreak.getM_Product_Category_ID() > 0)
+		{
+			schemaBreak.setM_Product_Category_ID(0);
+		}
+
+		enforcePriceLimit(schemaBreak);
+	}
+
+	public void enforcePriceLimit(final I_M_DiscountSchemaBreak schemaBreak)
+	{
+		if (!schemaBreak.isActive())
+		{
+			return;
+		}
+
+		final int productId = schemaBreak.getM_Product_ID();
+		if (productId <= 0)
+		{
+			return;
+		}
+
+		final PriceLimitEnforceContext context = PriceLimitEnforceContext.builder()
+				.schemaBreak(schemaBreak)
+				.isSOTrx(true)
+				.build();
+
+		Stream.of(context)
```

Comment:
@metas-mk do we need to enforce the price limit for all BPArtnerLocation-Countries? 
Maybe we just need to enforce it for the countries of "ShipTo"-Locations. WDYT?